File: backend/app/schemas/kb.py
from pydantic import BaseModel, Field


class KbQueryRequest(BaseModel):
    query: str = Field(min_length=1, max_length=2000)
    top_k: int = Field(default=5, ge=1, le=20)


class KbCitation(BaseModel):
    chunk_id: int
    question_id: int
    excerpt: str
    source_type: str


class KbQueryResponse(BaseModel):
    answer: str
    citations: list[KbCitation]


class KbReindexResponse(BaseModel):
    questions_processed: int

File: frontend/src/api/kb.ts
import { apiClient } from "./client";

export interface KbCitation {
  chunk_id: number;
  question_id: number;
  excerpt: string;
  source_type: string;
}

export interface KbQueryResponse {
  answer: string;
  citations: KbCitation[];
}

export async function postKbQuery(query: string, topK = 5): Promise<KbQueryResponse> {
  const res = await apiClient.post<KbQueryResponse>(
    "/api/kb/query",
    { query: query.trim(), top_k: topK },
    { timeout: 120000 }
  );
  return res.data;
}

export async function postKbReindex(): Promise<{ questions_processed: number }> {
  const res = await apiClient.post<{ questions_processed: number }>("/api/kb/reindex");
  return res.data;
}
